src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java

```diff
@@ -366,21 +357,19 @@ public ResultMessage execute(QueryState state, QueryOptions options, long queryS
 
         try
         {
+            // check again since now we have query options; note that statements are quaranted to be single partition reads at this point
             for (NamedSelect assignment : assignments)
-                checkAtMostOneRowSpecified(state.getClientState(), options, assignment.select, INCOMPLETE_PRIMARY_KEY_LET_MESSAGE);
+                checkFalse(isSelectingMultipleClusterings(assignment.select, options), INCOMPLETE_PRIMARY_KEY_SELECT_MESSAGE, "LET assignment", assignment.select.source);
 
             if (returningSelect != null)
-                checkAtMostOneRowSpecified(state.getClientState(), options, returningSelect.select, INCOMPLETE_PRIMARY_KEY_SELECT_MESSAGE);
+                checkFalse(isSelectingMultipleClusterings(returningSelect.select, options), INCOMPLETE_PRIMARY_KEY_SELECT_MESSAGE, "returning SELECT", returningSelect.select.source);
```

Comment:
I'd even be fine if we just did what we do for INSERT/UPDATE...that is, just say `SELECT at [2:1]` or `LET at [1:1]`. Not sure if someone reading this would glean information from "returning" or it's just an implementation detail.

CC @dcapwell

Reply:
"returning select" I think helps as we can only have 1 of those

Reply:
The only other SELECTs would be called LET here...anyway...I'm good with whatever the two of you like better.